### DISCUSSION TITLE
我们的认证鉴权逻辑比较复杂，内置的插件难以支持，如何配置 Higress 去调用某个接口（API）来完成认证鉴权的逻辑？

### DISCUSSION THREAD

**Discussion by jaggerwang** [higress-group/higress#1087] [Q&A] [answered]
只有接口返回状态为 200，并且响应体 JSON 的 code 属性值为 OK 才通过。只有自己开发插件这一条路吗？有没有一个通用的插件来完成 API 的调用并校验响应内容。

**Comment by CH3CHO** [comment]
感觉可以参考一下 Istio 的 External Authorization 功能：https://istio.io/latest/docs/tasks/security/authorization/authz-custom/

Higress 也支持用 Istio CRD 来进行配置的：https://higress.io/zh-cn/docs/ops/deploy-by-helm#%E6%94%AF%E6%8C%81-istio-crd%E5%8F%AF%E9%80%89

**Reply by jaggerwang** [reply]
没有启用 Istio，不想整这么复杂的东西。

**Reply by CH3CHO** [reply]
那还是得写个插件了。

**Reply by CH3CHO** [reply, marked as answer]
> 没有启用 Istio，不想整这么复杂的东西。

不需要安装 Istio 程序，只安装 CRD 就行。
